src/lightning_app/utilities/cluster_logs.py

```diff
@@ -0,0 +1,108 @@
+import json
+import queue
+from dataclasses import dataclass
+from threading import Thread
+from typing import Callable, Iterator, Optional
+
+import arrow
+import dateutil.parser
+from websocket import WebSocketApp
+
+from lightning_app.utilities.log_helpers import _error_callback, _OrderedLogEntry
+from lightning_app.utilities.logs_socket_api import _ClusterLogsSocketAPI
+from lightning_app.utilities.network import LightningClient
+
+
+@dataclass
+class _ClusterLogEventLabels:
```

Comment:
Have we thought about unifying that with AppLogs?  I think it should be doable ;D

Reply:
I general - yes, but we have to do quite a lot backend changes :)